enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java

```diff
@@ -230,6 +231,10 @@ public void obligation( long txId ) throws IOException
             {
                 obligationFulfiller.fulfill( txId );
             }
+            catch ( IllegalStateException e )
+            {
+                throw new ComException( "Failed to pull updates", e );
+            }
```

Comment:
Looks to me like the UpdatePuller is misusing the IllegalStateException. The state is rare but expected. The "strict" waiting mode could probably just wait past any pauses, unless there is some life support stuff requiring the UpdatePuller to be recreated.

Or false could be returned from await/fulfill and handled appropriately from clients, but still without any IllegalStateExceptions which are not even in the declaration of fulfill.